File: JScript/clsPfSection.js
//
// Copyright (c)2014 S C Harrison
// Refer to License.txt for terms and conditions of use.
//



//Define Class
function clsPfSection()
{
   key = 0;
  
   ax = 0;           //.. member's cross sectional area ..
   iz = 0;           //.. member's second moment of area ..
  
  //Dependent on Material Properties
   t_len = 0;        //.. TOTAL length of this section ..
   t_mass = 0;       //.. TOTAL mass of this section ..
   mat = 0;          //.. material of section ..
  
   Descr = "";       //.. section description string ..
}


clsPfSection.prototype.initialise = function()
{
  this.key = 0;
  this.ax = 0;
  this.iz = 0;
  this.mat = 0;
  this.Descr = "<unknown>";  
  
  this.t_len = 0;
  this.t_mass = 0;

}

clsPfSection.prototype.setValues = function(sectionKey, SectionArea, SecondMomentArea, materialKey, Description)
{
  this.key = sectionKey;
  this.ax = SectionArea;
  this.iz = SecondMomentArea;
  this.mat = materialKey;
  this.Descr = Description;
}

clsPfSection.prototype.sprint = function()
{
  var s;
  
  s = "";
  s = s + StrLPad(this.key.toString(), 8);
  s = s + StrLPad(this.ax.toString(), 15);
  s = s + StrLPad(this.iz.toString(), 15);
  s = s + StrLPad(this.mat.toString(), 6);
  s = s + StrLPad(this.Descr.toString(), 28);

return s;

}


clsPfSection.prototype.cprint = function()
{
  WScript.Echo(this.sprint());
}

clsPfSection.prototype.fprint = function(fp)
{
  fp.WriteLine(this.sprint());
}


clsPfSection.prototype.fgetData = function(fp)
{
  var s,s1;
  var n;
  var dataflds;
  var i,n;

  WScript.Echo( "fgetData ...");
  
   s = fp.ReadLine();
   //WScript.Echo( s);
   this.sgetData(s);

  WScript.Echo( "... fgetData");
}


clsPfSection.prototype.sgetData = function(s)
{
  var s1;
  var n;
  var dataflds;
  var i,n;

  WScript.Echo( "sgetData ...");
  WScript.Echo( s);

   dataflds = s.split(" ");
//   s = s.replace(/^\s+|\s+$/gm,''); //trim trailing spaces
//   dataflds = s.match(/-?\d+(?:[,.]\d+)?/gi);
   
   s1="";
   n = dataflds.length;
   for (i=0;i<n;i++)
   {
    WScript.Echo( i + "<" + dataflds[i] + ">" );
    if (dataflds[i] != "")
    {
      if(s1 == "") {
        s1 = dataflds[i];
      }else {
        s1 = s1 + "," + dataflds[i];
      }
    }
   }
   WScript.Echo(s1); 
   dataflds = s1.split(",");
   WScript.Echo();
   
   n = dataflds.length;
   for (i=0;i<n;i++)
   {
    WScript.Echo( i + "<" + dataflds[i] + ">" );
   }

  this.key = parseInt(dataflds[0]);
  this.ax = parseFloat(dataflds[1]);
  this.iz = parseFloat(dataflds[2]);
  this.mat = parseFloat(dataflds[3]);
  this.Descr = dataflds[4];
  
  //Zero Variables
  this.t_len = 0;
  this.t_mass = 0;

  WScript.Echo( "... sgetData");
}

File: SharpDevelop/CSharp/w7cpframe/clsPfSection.cs
//
// Copyright (c)2016 S C Harrison
// Refer to License.txt for terms and conditions of use.
//
using System;
using System.Collections.Generic;
using System.Text;
using System.IO;
using System.Text.RegularExpressions;


namespace w7cpframe
{
    class clsPfSection
    {

        //Define Class

        public int key = 0;

        public double ax = 0;           //.. member's cross sectional area ..
        public double iz = 0;           //.. member's second moment of area ..

        //Dependent on Material Properties
        public double t_len = 0;        //.. TOTAL length of this section ..
        public double t_mass = 0;       //.. TOTAL mass of this section ..

        public int mat = 0;          //.. material of section ..

        public string Descr = "";       //.. section description string ..



        public void initialise()
        {
            key = 0;
            ax = 0;
            iz = 0;
            mat = 0;
            Descr = "<unknown>";

            t_len = 0;
            t_mass = 0;

        }

        public void setValues(int sectionKey, double SectionArea, double SecondMomentArea, int materialKey, string Description)
        {
            key = sectionKey;
            ax = SectionArea;
            iz = SecondMomentArea;
            mat = materialKey;
            Descr = Description;
        }

        public string sprint()
        {
            string s;

            s = "";
            s = s + key.ToString().PadLeft(8);
            s = s + ax.ToString().PadLeft(15);
            s = s + iz.ToString().PadLeft(15);
            s = s + mat.ToString().PadLeft(6);
            s = s + Descr.ToString().PadLeft(28);

            return s;

        }


        public void cprint()
        {
            System.Console.WriteLine(sprint());
        }

        public void fprint(StreamWriter fp)
        {
            fp.WriteLine(sprint());
        }


        public void fgetData(StreamReader fp)
        {
            string s;


            System.Console.WriteLine("fgetData ...");

            s = fp.ReadLine();
            //WScript.Echo( s);
            sgetData(s);

            System.Console.WriteLine("... fgetData");
        }


        public void sgetData(string s)
        {
            string [] dataflds = new string[10]; //(0 To 9);
            char [] delimters = new char[] { ' ', '\t' };
            int i,n;
            string s1;

            System.Console.WriteLine("sgetData ...");
            System.Console.WriteLine(s);

            string regResult;
            string pattern = @"^\s+|\s+$"; //trim trailing spaces
            regResult = Regex.Replace(s, pattern, "");

            dataflds.Initialize();
            string pattern2 = @"-?\d+(?:[,.]\d+)?";
            i = 0;
            foreach (Match match in Regex.Matches(regResult, pattern2))
            {
                if (match.Value != "")
                {
                    dataflds[i] = match.Value;
                    System.Console.WriteLine(match.Value);
                    i++;
                }
            }

            s1 = "";
            n = dataflds.Length;
            for (i = 0; i < n; i++)
            {
                System.Console.WriteLine(i + "<" + dataflds[i] + ">");
                if (dataflds[i] != "")
                {
                    if (s1 == "")
                    {
                        s1 = dataflds[i];
                    }
                    else
                    {
                        s1 = s1 + "," + dataflds[i];
                    }
                }
            }
            System.Console.WriteLine(s1);
            System.Console.WriteLine();

            n = dataflds.Length;
            for (i = 0; i < n; i++)
            {
                System.Console.WriteLine(i + "<" + dataflds[i] + ">");
            }

            key = int.Parse(dataflds[0]);
            ax = double.Parse(dataflds[1]);
            iz = double.Parse(dataflds[2]);
            mat = int.Parse(dataflds[3]);
            Descr = dataflds[4];

            //Zero Variables
            t_len = 0;
            t_mass = 0;

            System.Console.WriteLine("... sgetData");
        }





    } //class
} // name space
